web/elm/src/Build/Header/Header.elm

```diff
@@ -177,6 +270,37 @@ tooltip model session =
                 , containerAttrs = Nothing
                 }
 
+        HoverState.Tooltip (BuildTab id _) _ ->
+            List.head (List.filter (\b -> b.id == id) model.history)
+                |> Maybe.andThen
+                    (\b ->
+                        let
+                            lines =
+                                String.lines b.comment
+                        in
+                        if String.isEmpty b.comment then
+                            Nothing
+
+                        else
+                            List.head lines
+                                |> Maybe.map
+                                    (\text ->
+                                        let
+                                            truncatedText =
+                                                if List.length lines > 1 || String.length text > 50 then
+                                                    String.left 50 text ++ "…"
+
+                                                else
+                                                    text
+                                        in
+                                        { body = Html.text truncatedText
+                                        , attachPosition = { direction = Tooltip.Bottom, alignment = Tooltip.Start }
+                                        , arrow = Nothing
+                                        , containerAttrs = Nothing
+                                        }
+                                    )
```

Comment:
Hmm, I thought about that, but then you lose the wrapping as the tooltip approaches the side of the window, like this...

![image](https://user-images.githubusercontent.com/354845/122663541-85a41500-d1de-11eb-8b1a-98ebd333175a.png)

We could avoid this by swapping from `alignment = Tooltip.Start` to `alignment = Tooltip.End` when we think that a tooltip would overflow the edge of the screen. I'll have a play, but can you think of any way to do this off the top of your head?

Also regarding the multiple lines, I did like the idea that only the first line of a comment would be shown which would make it kind of a heading, but happy to change if you think otherwise.

Reply:
In the latest commit I've set it to show the first line of text, truncated by CSS, without worrying about handling when the tooltip overflows the screen. I've left that out for now unless we can think of a nice way to do it.

I had a crack by passing the max dimensions into the tooltip view code and adding logic in there to set the `max-width` and `max-height` property, then ensuring that the `div` always stays within the bounds of the screen. This might work but I wanted to wrap up the other changes first so I'll come back to it later.

Reply:
Ah gotcha. To be honest I wouldn't worry too much about that as it currently impacts the rest of the tooltips in the UI. Maybe we can leave fixing tooltips overflowing over the edge of the window for a separate PR?